main.py

```diff
@@ -117,8 +119,8 @@ def write_candidate(perm: Permuter, source: str) -> None:
 
 def main(options: Options) -> int:
     last_time = time.time()
-    if options.seed is not None:
-        random.seed(options.seed)
+    if options.seed is None:
+        options.seed = os.urandom(8)
```

Comment:
this is bugging me for two reasons:
- options.seed is typed as an Optional[int], but here we assign bytes
- further down we check `if iteration == 0 and options.seed is not None`; now `options.seed is not None` cannot happen

Maybe we can type `options.seed` as int to resolve the second point and make the caller do the random seed defaulting work, if the goal is to have wrapped_main not depend on global randomness state? Otherwise you can move this seeding into that function.

Reply:
Yeah, this should be better now.

Reply:
Give me one sec, need to fix the initial seed setter

Reply:
Better now. Each permuter keeps track of its own current seed for each iteration. When initialized, when initialized every permuter starts with the same seed.

Reply:
(I also tested it this time, and get the same score updates)